File: auth-service/src/main/java/com/eirsteir/coffeewithme/authservice/repository/RoleRepository.java
package com.eirsteir.coffeewithme.authservice.repository;

import com.eirsteir.coffeewithme.authservice.domain.Role;
import com.eirsteir.coffeewithme.authservice.domain.RoleType;
import java.util.Optional;
import org.springframework.data.repository.CrudRepository;

public interface RoleRepository extends CrudRepository<Role, Long> {

  Optional<Role> findByType(RoleType type);
}

File: auth-service/src/main/kotlin/com/eirsteir/coffeewithme/authservice/repository/RoleRepository.kt
package com.eirsteir.coffeewithme.authservice.repository

import com.eirsteir.coffeewithme.authservice.domain.Role
import com.eirsteir.coffeewithme.authservice.domain.RoleType
import org.springframework.data.repository.CrudRepository
import java.util.*

interface RoleRepository : CrudRepository<Role, Long> {
    fun findByType(type: RoleType): Role?
}